js/ui/grid_core/ui.grid_core.header_filter_core.js

```diff
@@ -360,6 +357,28 @@ export const HeaderFilterView = modules.View.inherit({
         }
     },
 
+    _normalizeHeaderFilterOptions(options) {
+        const generalHeaderFilter = this.option('headerFilter') || {};
+        const specificHeaderFilter = options.headerFilter || {};
+
+        const generalDeprecated = {
+            search: {
+                enabled: generalHeaderFilter.allowSearch,
+                timeout: generalHeaderFilter.searchTimeout,
+            }
+        };
+
+        const specificDeprecated = {
+            search: {
+                enabled: specificHeaderFilter.allowSearch,
+                mode: specificHeaderFilter.searchMode,
+                timeout: specificHeaderFilter.searchTimeout
+            }
+        };
+
+        return extend(true, {}, generalHeaderFilter, generalDeprecated, specificHeaderFilter, specificDeprecated);
```

Comment:
Question.

Can we use the spread shell copy operator instead of the ```extend``` here?

E.g
```suggestion
        return {
          ...generalHeaderFilter,
          ...generalDeprecated,
          ...specificHeaderFilter,
          ...specificDeprecated,
        };
```

Reply:
Unfortunately, no. We need deepExtend here, because we have headerFilter.search option which is object. So if in `generalHeaderFilter.search = { enabled: true }` and `specificHeaderFilter.search = { editorOptions: { placeholder: 'some' } }`, then the result headerFilter will contain just `specificHeaderFilter.search` values